app/src/main/java/org/simple/clinic/storage/Timestamps.kt

```diff
@@ -0,0 +1,42 @@
+package org.simple.clinic.storage
+
+import androidx.room.Embedded
+import org.simple.clinic.util.UtcClock
+import org.threeten.bp.Instant
+
+/**
+ * There are three timestamps that are always present in every single synced model:
+ *
+ * - `createdAt`: When the database record was created.
+ * - `updatedAt`: When the database record was updated.
+ * - `deletedAt`: When the database record was deleted.
+ *
+ * All timestamps are in UTC.
+ *
+ * This class is meant to encapsulate these timestamps as an [Embedded] model in the database
+ * entities.
+ **/
+data class Timestamps(
```

Comment:
At the moment, we can have upto four different timestamps on any domain model: `createdAt`, `updatedAt`, `deletedAt`, and `recordedAt`. They are all of the same type, `Instant. 

However, if we look at them from a domain perspective, only `recordedAt has an actual meaning at the domain level (When did this patient visit? When was this blood pressure recorded?). The others are all related to the application logic (when was this database record created? When was it changed? Was it ever deleted?).

This was causing trouble when working with records because it is up to us to remember what fields to change when something happens and it is easily missable. I've been thinking about trying to introduce a `Timestamps` notion for a while now (https://www.pivotaltracker.com/story/show/162636638) which will separate these two concepts cleanly. Decided to try it out here.